borg/helpers.py

```diff
@@ -269,15 +269,32 @@ def list_archive_infos(self, sort_by=None, reverse=False):
         return archives
 
 
-def prune_within(archives, within):
+def within_range(s):
+    """Convert a string representing a valid 'within' range to a number of hours."""
     multiplier = {'H': 1, 'd': 24, 'w': 24 * 7, 'm': 24 * 31, 'y': 24 * 365}
+
+    if s.endswith(tuple(multiplier.keys())):
+        number = s[:-1]
+        suffix = s[-1]
+    else:
+        # range suffixes in ascending multiplier order
+        ranges = [ k for k, v in sorted(multiplier.items(), key=lambda t: t[1]) ]
+        raise argparse.ArgumentTypeError(
+            'Unexpected --keep-within suffix "%s": expected one of %s' % (s[-1], ranges))
```

Comment:
this is also somehow misphrased for here, similar to above issue.

maybe say 'Invalid time unit: expected one of ...'